src/main/java/org/jenkinsci/plugins/ewm/actions/ExwsAllocateActionImpl.java

```diff
@@ -53,6 +61,31 @@ public String getUrlName() {
         return "exwsAllocate";
     }
 
+    @Restricted(NoExternalUse.class)
+    public DirectoryBrowserSupport doWs(StaplerRequest req, StaplerResponse rsp, @QueryParameter String id) throws IOException, ServletException, InterruptedException {
+        ExternalWorkspace exws = null;
+
+        // TODO makes sense to make #allocatedWorkspaces a HashMap
+        for (ExternalWorkspace allocatedWorkspace : allocatedWorkspaces) {
```

Comment:
I would decouple this code to the temporary <code> ExternalWorkspace locateWorkspace(String id)</code>